pydantic/type_adapter.py

```diff
@@ -128,6 +128,17 @@ def __new__(cls, __type: Any, *, config: ConfigDict | None = ...) -> TypeAdapter
             """A class representing the type adapter."""
             raise NotImplementedError
 
+        @overload
+        def __init__(self, type: type[T], *, config: ConfigDict | None = None, _parent_depth: int = 2) -> None:
+            ...
+
+        # this overload is for non-type things like Union[int, str]
+        # Pyright currently handles this "correctly", but MyPy understands this as TypeAdapter[object]
+        # so an explicit type cast is needed
+        @overload
+        def __init__(self, type: T, *, config: ConfigDict | None = None, _parent_depth: int = 2) -> None:
```

Comment:
```suggestion
        def __init__(self, type: Any, *, config: ConfigDict | None = None, _parent_depth: int = 2) -> None:
```
Like `__new__` overload

Reply:
You are right